providers/implementations/encode_decode/encode_key2any.c

```diff
@@ -201,11 +201,12 @@ static int key_to_epki_der_priv_bio(BIO *out, const void *key,
         return 0;
 
     p8 = key_to_encp8(key, key_nid, str, strtype, k2d, ctx);
-    if (p8 != NULL)
+    if (p8 == NULL) {
+        free_asn1_data(strtype, str);
```

Comment:
Ah, I now see the reason why it was not called here but inside key_to_encp8(). If p8info_to_encp8() fails then there is a doublefree.

Reply:
Ok, I undid that